backend/index.ts

```diff
@@ -21,35 +21,57 @@ try {
 
 // Start the extension server and the metrics server
 extensionServer.listen(SOCKETFILE, () => {
-  console.log(`🚀 Server listening on ${SOCKETFILE}`);
+  console.log(`🚀  Server listening on ${SOCKETFILE}`);
 });
 
 metricsServer.listen(METRICS_PORT, () => {
-  console.log(`📈 Metrics are available on ${METRICS_PORT}/metrics`);
+  console.log(`📈  Metrics are available on ${METRICS_PORT}/metrics`);
 });
 
 (async () => {
   try {
     // Wait 15 seconds to give the Grafana container extra time to spin up and be ready to respond
     // to requests. This implementation can be improved later.
-    console.log('⏳ Waiting ... ');
-    await new Promise((r) => setTimeout(r, 1000 * 15));
-    console.log('⌛ Done waiting.');
+    console.log('⏳  Waiting ... ');
+    await new Promise((r) => setTimeout(r, 1000 * 10));
+    console.log('⌛  Done waiting.');
 
     // Get a list of all the Docker containers
     const response = await axios.get('/containers/json', {
       socketPath: DOCKER_DAEMON_SOCKET_PATH,
       params: { all: true },
     });
-    const containers = (response.data as DockerContainer[]).map(({ Id }) => Id);
 
-    // ###########################################################
-    // # TODO: Create Grafana dashboards for each container here #
-    // ###########################################################
+    // Populate two arrays, mapping through them in order ensures that their INDEX VALUES
+    // can properly CORRELATE each container's respective IDs and Names.
+    const containerIDs = (response.data as DockerContainer[]).map(({ Id }) => Id);
+    const containerNames: (string | undefined)[] = (response.data as DockerContainer[]).map(
+      ({ Names }) => Names.pop()?.slice(1)
+    );
+
+    // Get necessary datasource information from Grafana directly.
+    const datasource = await getGrafanaDatasource();
+
+    // Iterate through ID array and create a dashboard object for each container.
+    containerIDs.forEach(async (id, index) => {
+      const dashboard = await createGrafanaDashboardObject(id, containerNames[index], datasource);
```

Comment:
NIT: Here are you iterating through the containerIDs array, and within it you are accessing the container's name with `containerNames[index]`

Instead of creating two separate arrays, `containerIDs` and `containerNames`, you could just use the `containers` array that is given by the response.

Then, you could iterate through that `containers` array instead. I think that would be better than creating two additional arrays, and it would improve readability.

```js
containers.forEach((container) => {
  const containerId = container.Id;
  const containerName = container.Names[0].replace(/^\//, '');

  // Post request to Grafana API to create a dashboard using the returned dashboard object.
  // ...

});
```

You could use this strategy below too on the forEach function which creates the HTTP requests to Docker Engine